contracts/BasicPolyptychRandomizerV0.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Creatd By: Art Blocks Inc.
+
+pragma solidity 0.8.17;
+
+import "./interfaces/0.8.x/IBasicPolyptychRandomizerV0.sol";
+import "./interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "./interfaces/0.8.x/IFilteredMinterHolderV1.sol";
+
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+
+/**
+ * @title Basic Polytych Randomizer contract that enables the copying of token hash seeds.
+ * @notice This contract requires several configuration options to be set properly before
+ * it enables the copying of token hashes from an original token (the first panel of a polyptych)
+ * to subsequent diptych, triptych, or polyptych panels.
+ *
+ * Requirements to use the polyptych randomizer:
+ * - Core contract must be an instance of `GenArt721CoreV3_Engine`
+ * - The initial project used as the first panel must be minted with a completely random token hash
+ * - An `MinterPolyptychV0` contract or similar must be deployed and configured as the `hashSeedSetterContract`
+ * - The `projectDuplicateHashSeedLimit` should be configured for each project
+ *
+ * Once the contract is configured, it will enable newly-minted tokens to use a copy of the hash
+ * seed from a previously-minted token.
+ */
+contract BasicPolyptychRandomizerV0 is IBasicPolyptychRandomizerV0, Ownable {
+    // The core contract that may interact with this randomizer contract.
+    IGenArt721CoreContractV3 public genArt721Core;
+
+    // Used to obtain the project ID from the token ID
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    // Mapping of token IDs to hash seeds
+    mapping(uint256 => bytes12) public polyptychHashSeeds;
+
+    // Mapping of the number of times a hash seed has been used
+    mapping(bytes12 => uint) public hashSeedUsageCounter;
+
+    // Configures the maximum number of hash seed copies that can exist per project (default zero)
+    mapping(uint256 => uint) public projectDuplicateHashSeedLimit;
```

Comment:
Instead of being a boolean mapping, this keeps track of the number of times we allow a project's token hash to be copied.

This limit is set per-project to allow two different use cases:

1. If polyptych panels are all part of the same project, the artist can increment this maximum limit (default zero) to allow the next panel to be minted on the same project ID
2. If polyptych panels are in a new project, the new project should have a limit that is equal to the previous number of panels

Reply:
Consider the following "panels in separate projects" (use case 1 above):
- Base Panel A: project 0
- Panel B: project 1
- Panel C: project 2
- Artist sets `projectDuplicateHashSeedLimit[1]` to 1, `projectDuplicateHashSeedLimit[2]` 2
- If Panel C is minted before Panel B:
  - Panel B can never be minted, because the randomizer has a duplicate hash seed limit of 1 for project 1
  - Panel C can be minted twice with the same token hash
  
Both of those last two bullets are unintended behaviors. To avoid, we essentially require that EVERY token is minted on Panel B's project before minting on C can be opened up, or require that an artist perform the mints via their privileged account while the project is paused, and mint in the intended order of Panel B, Fame C.

I also have some reservation with defining this on the randomizer, mostly in the sense that this feels a bit like ~minter state being tracked on the randomizer. If a new randomizer were used in the future, it would have to look up `hashSeedUsageCounter` and `projectDuplicateHashSeedLimit` values on the old minter based on some kind of logic, for any ongoing projects, which is possible, but is a bit gas inefficient.

Based off of last week's discussion, was thinking we would implement something like the following on the minter itself:
```
// mapping tracks which base tokens have been used in project's current panel
// panelId => baseTokenAddress => baseTokenId => panelIsMinted
mapping(uint256 => mapping(address => mapping(uint256 => bool))) panelIsMinted;
// mapping that identifies a project's current panel id
mapping(uint256 => uint256) projectToCurrentPanelId;

...

// function that allows artist to increment panel id
function incrementPanelId(_projectId) ... {
    ...
    projectToCurrentPanelId[_projectId]++;
}

...

// somewhere within purchase function...
uint256 currentPanelId = projectIdToCurrentPanelId[_projectId]
require(!panelIsMinted[currentPanelId][baseTokenAddress][baseTokenId], "Panel already minted));
!panelIsMinted[currentPanelId][baseTokenAddress][baseTokenId] = true;
...
```

That kind of solution ends up constraining only the mint-order of projects with multiple panels in the same project, in that it requires panel B be minted before panel C if we want panel B to exist. However, due to the nature of tokenIds and artist scripts, this is very likely already a constraint since a project's script must have a sense of which panel is being minted a-priori (e.g. Panel A is tokens 0 through 49, Panel B is tokens 50 through 99, etc.). 

Apologies for the long comment 😅 thoughts?